src/components/colliders/BaseCollider.vue

```diff
@@ -0,0 +1,61 @@
+<script setup lang="ts">
+import { inject, onUnmounted, type ShallowRef, shallowRef, watch } from 'vue'
+
+import { useRapierContext } from '../../composables'
+import { createCollider } from '../../utils/collider'
+import type { ColliderProps, CreateColliderReturnType, RigidBodyContext } from '../../types'
+
+const props = withDefaults(defineProps<Partial<ColliderProps>>(), {
+  shape: 'cuboid',
+  args: () => [1, 1, 1],
+})
+
+const { world } = useRapierContext()
+
+const bodyContext = inject<ShallowRef<RigidBodyContext>>('bodyContext') ?? shallowRef<RigidBodyContext>()
+const colliderInfos = shallowRef<CreateColliderReturnType>()
+const instance = shallowRef<CreateColliderReturnType['collider']>()
+const instanceDesc = shallowRef<CreateColliderReturnType['colliderDesc']>()
+
+defineExpose({
+  instance,
+  colliderDesc: instanceDesc,
+})
+
+// TODO: Correctly wait for the physics state to be available
+watch(bodyContext, (state) => {
```

Comment:
Have you tried nextTick?

Something like:
watch(bodyContest, async  (state) => {
await nextTick()
...

Reply:
Let's give it a shot

Reply:
I tested and worked for the first render, but when trying to update values and triggering the re-render, I'm getting the same behavior

Reply:
I think to really be able to solve this issue, we have to listen to the node update, and apply the new object position to the collider

Reply:
Let me know when you solve, and please don't get hurry. Nice weekend 😊

Reply:
Yeah sure, probably tomorrow I'll be able to solve this,
Thanks, Nice weekend too!